src/components/BankDetails/BankDetails.tsx

```diff
@@ -14,39 +14,64 @@ const DEFAULT_EXPECTED_MONTHLY_DONATIONS_AMOUNT = "1000";
 
 type Props = {
   FormButtons: IFormButtons;
+  /** All errors should be handled inside `onSubmit` */
   onSubmit: OnSubmit;
 };
 
 export default function BankDetails({ FormButtons, onSubmit }: Props) {
-  const [currency, setCurrency] = useState<Pick<WiseCurrency, "code" | "name">>(
-    { code: "USD", name: "United States Dollar" }
-  );
+  const [isSubmitting, setSubmitting] = useState(false);
+  const [currency, setCurrency] = useState<Currency>({
+    code: "USD",
+    name: "United States Dollar",
+  });
   const [amount, setAmount] = useState(
     DEFAULT_EXPECTED_MONTHLY_DONATIONS_AMOUNT
   );
   const [debouncedAmount] = useDebouncer(amount, 500);
   const amnt = /^[1-9]\d*$/.test(debouncedAmount) ? +debouncedAmount : 0;
 
+  const handleSubmit: OnSubmit = async (...params) => {
+    try {
+      setSubmitting(true);
+      await onSubmit(...params);
+    } catch (error) {
+      // All errors should be handled in `onSubmit`.
+      // This try/catch is just to ensure that `isSubmitting`
+      // is set to false at the end of the operation.
+      console.error(error);
+    } finally {
+      setSubmitting(false);
```

Comment:
comment: I didn't stress too much on disabling currency and amount as user is way below the dynamic form when they hit submit - and probably won't gain any benefit changing mid-submit if they can

Reply:
Understandable. I think of this more as a UX aspect, to indicate to the user they shouldn't be touching anything while the form is submitting or loading.